### PR TITLE
cudos-cli account management

### PR DESCRIPTION
**Pull request by varasu**

### CODE REVIEW

File: cmd/keys.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                    describe: 'amount of tokens in the format 10000000ucudos',
+                })
+        }, fund)
+        .command('ls', 'List all accounts in the keystore', () => {}, list)
+        .command('rm [name]', 'Remove account from the keystore', () => {
+            yargs.positional('name', {
+                    type: 'string',
+                    describe: 'account name',
+                }),
+                yargs.option('yes', {
```
Comment (Infra1515) [inline cmd/keys.js:82]: maybe rename the flag to --no-confirmation so it would be a bit more explicit what it is doing

File: cmd/lib/keypair.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+const bip39 = require('bip39');
+const bip32 = require('bip32');
+const secp256k1 = require('secp256k1');
+const {
+    bech32
+} = require('bech32');
+const createHash = require('create-hash');
+
+function createFromMnemonic(mnenonic, hdPath) {
+    if (typeof mnemonic === 'undefined') {
```
Comment (Infra1515) [inline cmd/lib/keypair.js:10]: argument is different from checked variable
createFromMnemonic(mnenonic, hdPath)
mnemonic === 'undefined']

**mnenonic != mnemonic**

File: cmd/lib/keypair.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    } = masterKey.derivePath(hdPath)
+    return privateKey;
+}
+
+function createKeyPair(mnemonic, hdPath = `m/44'/118'/0'/0/0`) {
+    return createFromMnemonic(mnemonic, hdPath);
+}
+
+module.exports = {
+    Create: createKeyPair,
```
Comment (Infra1515) [inline cmd/lib/keypair.js:51]: createKeyPair takes two arguments but none is passed ( only hdPath with default value ) -  mnemonic will be added in the future ?

File: cmd/lib/keystore.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    async init() {
+        if (!fsExtra.pathExists(this.keyStoreDir)) {
+            fsExtra.mkdirp(this.keyStoreDir)
+        }
+    }
+
+    async getAccountPath(name) {
+        const accountPath = path.join(this.keyStoreDir, name)
+        if (!await fsExtra.pathExists(accountPath)) {
```
Comment (Infra1515) [inline cmd/lib/keystore.js:26]: handle the exception gracefully

File: cmd/lib/keystore.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        return {
+            mnemonic: kp.mnemonic,
+            address: keypair.getAddressFromPrivateKey(kp.privateKey)
+        };
+    }
+
+    async saveAccount(name, account) {
+        const accountPath = path.join(this.keyStoreDir, name)
+        if (await fsExtra.pathExists(accountPath)) {
+            throw new Error('Account already exists in the keyStore.');
```
Comment (Infra1515) [inline cmd/lib/keystore.js:46]: same, handle the exception in a user friendly way

File: cmd/keys.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+const add = async function(argv) {
+    const acc = await keystore.createNewAccount(argv.name);
+    console.log(`Account ${argv.name} is created. ${acc.address}`);
+    console.log(`Keep the mnemonic in a secure location. This is the only way to recover your account.\n${acc.mnemonic}`)
+}
+
+
+const fund = async function(argv) {
+    if (typeof argv.name !== 'undefined') {
```
[the remark this answers is not in the document]
Comment (varasu) [reply cmd/keys.js:37]: It is a good practice to use typeof for this kind of checking.

**Review by rdpnd — APPROVED** [review]

File: cmd/lib/contract.js [hunk 7]
```diff
@@ -58,7 +67,7 @@ const Contract = class {
 
     async init() {
         if (!this.deployed) {
-            let artifacts = path.join(process.cwd(), `contracts/${this.contractname}/artifacts`);
+            let artifacts = path.join(process.cwd(), `/artifacts`);
             let r = await FindFiles(artifacts, /\.wasm$/);
             if (r.length < 1 || this.deployed) {
                 console.log('Use [compile] command to compile all the contracts in the project.');
```
[the remark this answers is not in the document]
Comment (mlukanova) [reply cmd/lib/contract.js]: Actually with the current structure in the init project this is a very likely scenario. We have alpha and beta contracts that are compiled in the same artefacts dir.

**Pull request merged**